### ISSUE TITLE
error: failed to get cargo metadata: No such file or directory(os error 2)

### ISSUE THREAD

**Issue by 2eppeli** [gatomod/tcp-chat#2]
sorry new error occured during “tauri 'build' ” after "pnpm tauri build". 
how to solve it?

**Comment by gatomod** [comment]
Please, could you describe more your problem? It would be nice to have the steps to reproduce it and logs.
Using "pnpm tauri build" should work, I tried it on different platforms and works. Did you installed all dependencies?

**Comment by 2eppeli** [comment]
![image](https://github.com/gatomod/tcp-chat/assets/45452562/ef9017a0-97e5-4e32-876f-6b992fa2cb99)
![image](https://github.com/gatomod/tcp-chat/assets/45452562/fac56053-ecbd-4b59-8ef3-82141e3f8df8)
yeah,  are these steps correct? i think i've installed all dependencies.
the platform is ubuntu23 desktop.

**Comment by 2eppeli** [comment]
and it's the last step i typed before the error
![image](https://github.com/gatomod/tcp-chat/assets/45452562/1bacc850-644e-4603-927f-afe39e559aa7)

**Comment by gatomod** [comment]
I think pkg-config doesn't refers to an APT package, also I don't recommend you to run sudo for that kind of things for security reasons. pnpm tauri build should work. I was researching about that error and looks that building Tauri apps in Linux is a bit complicated due to Glibc compatibility.
https://tauri.app/v1/guides/building/linux#limitations
https://github.com/tauri-apps/tauri/issues/1355
https://github.com/tauri-apps/tauri/issues/1355#issuecomment-824697825
https://github.com/tauri-apps/tauri/issues/1838 (same problem as you)

It's weird that you receive that errors if you have the latest Ubuntu version. I haven't had any similar problem when I built it. The only thing that I can recommend you is to update your system and Rust (via "rustup", `rustup update`). If after that is still not working, try to use a nightly Rust release (`rustup default nightly`).
I hope that issues I linked help you.
